changelog.d/10208.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in v1.35.1 where an `allow` key of a `m.room.join_rules` event could be applied for incorrect room versions and configurations.
```

Comment:
I'm not sure this explains it well enough, but not awaiting there it means that you could have a room with `allow` set in the join rules, but a join rule of *not* `restricted` and those allow rules would still be applied.

Realistically I don't think this is user visible unless someone was messing with join rules in a weird way, although it means you could slightly leak data if you did this.

Reply:
If this was more user-visible, I'd probably suggest rephrasing "an `allow` key of a `m.room.join_rules` event could be applied" with something more user-digestible, but as it's not it's likely more useful to be clear about what exactly the missing await affected. In which case, to a developer, this is very clear :slightly_smiling_face: